app/questionnaire/placeholder_parser.py

```diff
@@ -88,41 +69,23 @@ def _parse_transforms(self, transform_list: Sequence[Mapping]):
 
             for arg_key, arg_value in transform["arguments"].items():
                 if isinstance(arg_value, list):
-                    transform_args[arg_key] = self._resolve_value_source_list(arg_value)
+                    transformed_value = self._value_source_resolver.resolve(arg_value)
                 elif isinstance(arg_value, dict):
                     if "value" in arg_value:
-                        transform_args[arg_key] = arg_value["value"]
+                        transformed_value = arg_value["value"]
                     elif arg_value["source"] == "previous_transform":
-                        transform_args[arg_key] = transformed_value
+                        transformed_value = transformed_value
                     else:
-                        transform_args[arg_key] = self._resolve_value_source(arg_value)
+                        transformed_value = self._value_source_resolver.resolve(
+                            arg_value
+                        )
                 else:
-                    transform_args[arg_key] = arg_value
+                    transformed_value = arg_value
+
+                transform_args[arg_key] = escape_value(transformed_value)
```

Comment:
I think the escaping of the value should be done in one place (in the `_parse_placeholder` method), to reduce:
- the chance of it being missed with any future changes to logic
- the chance of double escaping as transforms can be chained

Reply:
For me to move the escaping to a level above, I would have to make one method for getting the transformed args and one for them getting the transformed value. Is that what you meant?

Reply:
No, the arguments don't need to be escaped, only the final value that is returned for the placeholder. I saw it as changing the `_parse_placeholder` method from:

```python
    def _parse_placeholder(self, placeholder: Mapping) -> Any:
        try:
            return self._parse_transforms(placeholder["transforms"])
        except KeyError:
            resolved_value = self._value_source_resolver.resolve(placeholder["value"])
            return escape_value(resolved_value)
```
to something like:
```python
    def _parse_placeholder(self, placeholder: Mapping) -> Any:
        try:
            value = self._parse_transforms(placeholder["transforms"])
        except KeyError:
            value = self._value_source_resolver.resolve(placeholder["value"])
        return escape_value(value)
```

Reply:
Ah, ya confused myself, I was thinking about escaping each argument instead of the end result. Will change.

Reply:
As discussed, the final return value cannot be escaped since we have transforms that add HTML such as `format_list` which would break.

Reply:
We will need to do something with the escaping. Currently it escapes the transform arguments, which could be the result of a previous transform. We either need to escape the return values from the value source resolver, or change the resolver to allow returning escaped values. 

As discussed, we will add a flag to the resolver to enable escaping answer values.

Reply:
Add a flag so that `ValueSourceResolver` can now escape answer values.

Reply:
As discussed, make changes to improve typing annotations. The `escape_value` method now lives in `answer.py` and is specific to our `Answer` implementation to make typing more explicit. This also allowed better type annotations for the resolver.